### PR TITLE
Bring vm/host_nexus.rb to 100% line/branch coverage

### PR DESCRIPTION
**Pull request by fdr**

### CODE REVIEW

File: prog/vm/host_nexus.rb [hunk 1]
```diff
@@ -36,16 +36,19 @@ def wait_prep
...
+        kwargs = {
+          total_sockets: st.dig(:exitval, "total_sockets"),
+          total_nodes: st.dig(:exitval, "total_nodes"),
+          total_cores: st.dig(:exitval, "total_cores"),
+          total_cpus: st.dig(:exitval, "total_cpus")
+        }
 
-        vm_host.update(total_sockets: total_sockets, total_nodes: total_nodes,
-          total_cores: total_cores, total_cpus: total_cpus)
+        fail "BUG: one of the LearnCores fields is not set" if kwargs.value?(nil)
```
Comment (furkansahin) [inline prog/vm/host_nexus.rb:49]: let's print `kwargs` here, it's easier to debug if we can see which value is nil from the error message directly.

File: spec/coverage_helper.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
   SimpleCov.start do
     enable_coverage :branch
-    minimum_coverage line: 87.99, branch: 73.64
+    minimum_coverage line: 88.2, branch: 80.18
```
Comment (furkansahin) [inline spec/coverage_helper.rb:8]: this is a good bump 🚀

File: spec/prog/vm/nexus_spec.rb [hunk 3]
```diff
@@ -14,7 +14,7 @@
   let(:vm) { Vm.new(size: "m5a.2x").tap { _1.id = "a410a91a-dc31-4119-9094-3c6a1fb49601" } }
   let(:tg) { TagSpace.create(name: "default").tap { _1.associate_with_tag_space(_1) } }
 
-  describe "#assemble" do
+  describe ".assemble" do
```
Comment (furkansahin) [inline spec/prog/vm/nexus_spec.rb:17]: Is this kind of a new convention that we want assemble function tests to start with `.` and states with`#`? If so, makes sense. If not and some kind of an extra change you didn't intend to do, I still like it 👍

**Review by furkansahin — APPROVED** [review]

File: spec/prog/vm/host_nexus_spec.rb [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  end
+
+  describe "#prep" do
+    it "starts a number of sub-programs" do
+      nx.instance_variable_set(:@vm_host, instance_double(VmHost,
+        net6: NetAddr.parse_net("2a01:4f9:2b:35a::/64")))
+      budded = []
+      expect(nx).to receive(:bud) do
+        budded << _1
+      end.at_least(:once)
```
Comment (enescakir) [inline spec/prog/vm/host_nexus_spec.rb:45]: I did not know that 👌

**Review by enescakir — APPROVED** [review]

File: spec/prog/vm/nexus_spec.rb (hunk 3 above)
Comment (fdr) [reply spec/prog/vm/nexus_spec.rb:17]: Oh yeah I looked it up, it is the convention for class methods, but it also was an extra change.

File: prog/vm/host_nexus.rb (hunk 1 above)
Comment (fdr) [reply prog/vm/host_nexus.rb:49]: Dynamic exception messages tend to mess up stacking in exception-aggregation software. We should probably do something to improve the logging situation.

**Pull request merged**